src/Piece.js

```diff
@@ -1,5 +1,35 @@
 /* Represents a piece in the game */
 class Piece 
 {
+	constructor()
+	{
+		this.moveVectors = [];
+		this.captureVectors = [];
+		this.moveCaptureVectors = [];
```

Comment:
Possibly rename moveCaptureVectors or captureVectors to be easier to remember. Right now they are somewhat easy to confuse for each other.

Reply:
Until such time as I can think of a better name, I'm ignoring this. I declare it a future me problem.

Reply:
I think the names are fine. I'm not worried about it.